allennlp/data/dataset_readers/coreference_resolution/winobias.py

```diff
@@ -0,0 +1,167 @@
+    on another corpora.
+
+    Returns a list of ``Instances`` which have four fields: ``text``, a ``TextField``
+    containing the full sentence text, ``spans``, a ``ListField[SpanField]`` of inclusive start and
+    end indices for span candidates, and ``metadata``, a ``MetadataField`` that stores the instance's
+    original text. For data with gold cluster labels, we also include the original ``clusters``
+    (a list of list of index pairs) and a ``SequenceLabelField`` of cluster ids for every span
+    candidate.
+
+    Parameters
+    ----------
+    max_span_width: ``int``, required.
+        The maximum width of candidate spans to consider.
+    token_indexers : ``Dict[str, TokenIndexer]``, optional
+        This is used to index the words in the sentence.  See :class:`TokenIndexer`.
+        Default is ``{"tokens": SingleIdTokenIndexer()}``.
+    """
+    def __init__(self,
+                 max_span_width: int,
+                 token_indexers: Dict[str, TokenIndexer] = None,
+                 lazy: bool = False) -> None:
+        super().__init__(lazy)
+        self._max_span_width = max_span_width
+        self._token_indexers = token_indexers or {"tokens": SingleIdTokenIndexer()}
+
+    @overrides
+    def _read(self, file_path: str):
+        # if `file_path` is a URL, redirect to the cache
+        file_path = cached_path(file_path)
+
+        for sentence in open(file_path, "r"):
+            tokens = sentence.strip().split(" ")
+            clusters: DefaultDict[int, List[Tuple[int, int]]] = collections.defaultdict(list)
+            words = []
+            inside_square_span = False
+            inside_round_span = False
+            for index, token in enumerate(tokens):
+                # Coreference is annotated using [square brackets]
+                # or (round brackets) around coreferent phrases.
+                if "[" in token and "]" in token:
+                    clusters[0].append((index, index))
+                elif "[" in token:
+                    clusters[0].append((index, index))
+                    inside_square_span = True
+                elif inside_square_span or "]" in token:
+                    old_span = clusters[0][-1]
+                    clusters[0][-1] = (old_span[0], old_span[1] + 1)
+                    inside_square_span = False
+
+                if "(" in token and ")" in token:
+                    clusters[1].append((index, index))
+                elif "(" in token:
+                    clusters[1].append((index, index))
+                    inside_round_span = True
+                elif inside_round_span or ")" in token:
+                    old_span = clusters[1][-1]
+                    clusters[1][-1] = (old_span[0], old_span[1] + 1)
+                    inside_round_span = False
+
+                if token.endswith("."):
+                    # Winobias is tokenised, but not for full stops.
+                    # We'll just special case them here.
```

Comment:
Does it already split things like "don't"?  And won't this behave poorly for things like "e.g."?

Reply:
the sentences are so simple, they don't contain things like this. Sorry I should have made clear how simplistic this data is in the docstring.

Reply:
And yes they already tokenise the few abbreviations they do have